README.rst

```diff
@@ -112,6 +112,146 @@ By default both minidump-stackwalk and breakpad are built with PGO and an
 appropriate training set is included in the sources.
 
 
+minidump-stackwalk output
+=========================
+
+
+JSON output
+-----------
+
+Rough JSON output schema:
+
+::
+
+  {
+    "status": <string>,                      // OK | ERROR_* | SYMBOL_SUPPLIER_INTERRUPTED
+    "system_info": {
+      "os": <string>,                        // Linux | Windows NT | Mac OS X
+      "os_ver": <string>,
+      "cpu_arch": <string>,                  // x86 | amd64 | arm | ppc | sparc
+      "cpu_info": <string>,
+      "cpu_count": <int>,
+      "cpu_microcode_version": <int>         // optional
+    },
+    "crash_info": {
+      "type": <string>,
+      "crash_address": <string>,             // 0x[[:xdigit:]]+, optional
+      "crashing_thread": <int>,              // thread index | null; optional
+      "assertion": <string>                  // optional
+    }
+    "largest_free_vm_block": <string>,       // 0x[[:xdigit:]]+
+    "lsb_release": {                         // this section is optional and covers Linux Standard Base information
+      "id": <string>,
+      "release": <string>,
+      "codename": <string>,
+      "description": <string>
+    },
+    "main_module": <int>,                    // index of module in modules list
+    "modules_contain_cert_info": true,       // optional
+    "modules": [
+      // zero or more
+      {
+        "base_addr": <string>,               // 0x[[:xdigit:]]+
+        "debug_file": <string>,              // filename | empty string
+        "debug_id": <string>,                // [[:xdigit:]]{33} | empty string
+        "end_addr": <string>,                // 0x[[:xdigit:]]+
+        "filename": <string>,
+        "code_id": <string>,
+        "version": <string>,
+        "loaded_symbols": true,              // optional, if mdsw looked for the file and it does exist
+        "missing_symbols": true,             // optional, if mdsw looked for the file and it doesn't exist
+        "corrupt_symbols": true,             // optional, if mdsw found a file that has parse errors
+        "symbol_disk_cache_hit": <bool>,     // optional, whether or not the SYM file was fetched from disk cache
+        "symbols_fetch_time": <float>,       // optional, time in ms it took to fetch symbol file from url; omitted
+                                             // if the symbol file was in disk cache
+        "symbol_url": <string>,              // optional, url of symbol file
+        "cert_subject": ?                    // optional
```

Comment:
I couldn't figure out what this was and I haven't seen examples in crash reports processed by Socorro.

@gabrielesvelto , @Gankra : Any ideas what this is?

Reply:
The `cert_subject` field contains the entity that signed a given module using Microsoft's Authenticode system. For our modules for example this is going to be "Mozilla Corporation", Microsoft own modules will be signed by "Microsoft Windows" or "Microsoft Corporation" and so on. The minidump-analyzer [retrieves this information](https://searchfox.org/mozilla-central/rev/98a9257ca2847fad9a19631ac76199474516b31e/toolkit/crashreporter/minidump-analyzer/minidump-analyzer.cpp#217-241) when it processes a crash and stuffs it in an annotation called [ModuleSignatureInfo](https://searchfox.org/mozilla-central/rev/98a9257ca2847fad9a19631ac76199474516b31e/toolkit/crashreporter/minidump-analyzer/minidump-analyzer.cpp#493).

Socorro's stackwalker uses this info to [populate that field](https://github.com/mozilla-services/minidump-stackwalk/blob/main/minidump-stackwalk/stackwalker.cc#L706-L709).